packages/models/src/types.ts

```diff
@@ -1,20 +1,66 @@
 /**
  *
  * @export
- * @type JSONValue
+ * @type Entitlements
  */
-export type JSONValue = string | number | boolean | { [x: string]: JSONValue } | Array<JSONValue>
+export type Entitlements = { [x: string]: string | number | boolean | Array<string | number | boolean> }
 
 /**
  *
  * @export
  * @type TokenGetter
  */
-export type TokenGetter = (forceRefresh: boolean) => Promise<string>
+export type TokenGetter = (forceRefresh?: boolean) => Promise<string>
 
 /**
  *
  * @export
  * @type EntitlementsCallback
  */
-export type EntitlementsCallback = (entitlements: JSONValue) => void
+export type EntitlementsCallback = (entitlements: Entitlements) => void
+
+/**
+ * Client credentials
+ */
+export interface ClientCredentials {
+  /**
+   * Planship API client ID
+   * @type {string}
+   */
+  clientId: string
+
+  /**
+   * Planship API client secret
+   * @type {string}
+   */
+  clientSecret: string
+}
+
+export function isClientCredentials(obj: any) {
+  return typeof obj?.clientId === 'string' && typeof obj?.clientSecret === 'string'
```

Comment:
Is this always sufficient or should we check length as well. Curious what happens if someone was to do something weird like clientId: "" etc.

Reply:
As discussed this should be changed to either check for the token function to avoid multiple obj/type checks or made more robust in general. Fine for now.